src/Controllers/DeleteProductController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+
+namespace App\Controllers;
+
+
+use App\Abstracts\Controller;
+use App\Interfaces\ProductModelInterface;
+use App\Validators\SkuValidator;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class DeleteProductController extends Controller
+{
+    private $productModel;
+
+    /**
+     * DeleteProductController constructor.
+     */
+    public function __construct(ProductModelInterface $productModel)
+    {
+        $this->productModel = $productModel;
+    }
+
+    public function __invoke(Request $request, Response $response, array $args)
+    {
+        $sku = $args['sku'];
+
+        $responseData = [
+            'success' => false,
+            'message' => '',
+            'data' => []
+        ];
+
+        try {
+            $sku = SKUValidator::validateSKU($sku);
+
+        } catch (\Throwable $e) {
+            $responseData['message'] = $e->getMessage();
+
+            return $this->respondWithJson($response, $responseData, 400);
+        }
+
+        try {
+            $exists = $this->productModel->checkProductExists($args['sku']);
+
+            if ($exists) {
+                $deleteProduct = $this->productModel->deleteProductBySku($args['sku']);
```

Comment:
you are passing the sku in from the args when you previously made the $sku variable. Its fine it works but its a bit weird